pkg/controllers/controller.go

```diff
@@ -54,22 +55,26 @@ func (c *GenericController) Reconcile(ctx context.Context, req reconcile.Request
 		provisioner.StatusConditions().MarkTrue(v1alpha1.Active)
 	}
 	// 4. Reconcile
-	if _, err := c.Controller.Reconcile(ctx, resource); err != nil {
+	if result, err := c.Controller.Reconcile(ctx, resource); err != nil {
 		zap.S().Errorf("Controller failed to reconcile kind %s, %s",
 			resource.GetObjectKind().GroupVersionKind().Kind, err.Error())
-		return reconcile.Result{Requeue: true}, nil
+		return result, err
 	}
 	// 5. Update Status using a merge patch
 	if err := c.Status().Patch(ctx, resource, client.MergeFrom(persisted)); err != nil {
-		return reconcile.Result{}, fmt.Errorf("Failed to persist changes to %s, %w", req.NamespacedName, err)
+		return reconcile.Result{Requeue: true}, fmt.Errorf("Failed to persist changes to %s, %w", req.NamespacedName, err)
 	}
-	return reconcile.Result{RequeueAfter: c.Interval()}, nil
+	// 6. Requeue controllers on an interval
+	if controllerType := reflect.TypeOf(c.Controller).String(); controllerType != "*termination.Controller" {
```

Comment:
Instead of this reflection, can we make the allocator controller simply return reconcile.Result{RequeueAfter: c.Interval()} and then this top level can just return the result directly?

Reply:
Good call here. I like this better, and generalizes better.